### ISSUE TITLE
Tabbing popup with autocomplete breaks on certain input

### ISSUE THREAD

**Issue by Ynfiniti** [skeletonlabs/skeleton#1586]
### Current Behavior

I have an autocomplete which shows as a popup whenever the user focusses the connected search input field `event: 'focus-click`
If the user does not type anything into the input field, but only focuesses it everything works fine. 
The user is able to select the wanted option from the autocomplete popup by using the `TAB` key.

After an input is done and the options are updated the popup will not gain any visible focus even when pressing the `TAB` key multiple times.
More Specifically, this failure only accures when the first entrie of all options is removed due to the new input. If however the user gives an input that would leave the previously first option in place, or if the user refocusses the input component, then tabbing works again.

_Another thing is, that the animations of removing/adding options from the list is broken._

As an example I will use the common example options from the skeleton website:
```typescript
let flavorOptions: AutocompleteOption[] = [
    { label: 'Vanilla', value: 'vanilla', keywords: 'plain, basic', meta: { healthy: false } },
    { label: 'Chocolate', value: 'chocolate', keywords: 'dark, white', meta: { healthy: false } },
    { label: 'Strawberry', value: 'strawberry', keywords: 'fruit', meta: { healthy: true } },
    { label: 'Neapolitan', value: 'neapolitan', keywords: 'mix, strawberry, chocolate, vanilla', meta: { healthy: false } },
    { label: 'Pineapple', value: 'pineapple', keywords: 'fruit', meta: { healthy: true } },
    { label: 'Peach', value: 'peach', keywords: 'fruit', meta: { healthy: true } }
];
``` 

At the beginning the autocomplete will show all options with the `Vanilla` option on the top. 
If the user now types the letter "v" for example, the options will update and tabbing will work just fine.
But if the input is "ch", then `Vanilla` will be removed from the top and now `Chocolate` is at the first index. 
You can press `TAB` as often as you want now, nothing will happen, but after refocussing the input field its fine again (until a new input ruins it)

### Expected Behavior

When the user gives a new input, the autocomplete component updates its options list (including all animations) and the user can now tab through the new set of options without having to refocus the component.

### Steps To Reproduce

1. Using this [stackblitz example](https://stackblitz.com/edit/skeletonlabs-repl-22yqvw?file=src/routes/+page.svelte) from another user (which uses skeleton@1.1.0)
2. You can try out everything inside the "Using different settings" card and things will work just as expected
3. Update Skeleton version using `pnpm update`
4. Go to `routes/+page.svelte` and change all occurances of `'focus'` into `'focus-click'` so it works with the new version
5. Use `pnpm run dev` to start the localhost with all new versions and give it some time
6. Try everything again

I will provide another stackblitz link as the reproduction link which is just a fork of the above one and setup until step 5 so you can immeadiately try out step 6

### Link to Reproduction / Stackblitz

https://stackblitz.com/edit/skeletonlabs-repl-p6rb8f?file=src%2Froutes%2F%2Bpage.svelte

### More Information

Things will be broken for the [documentation](https://www.skeleton.dev/components/autocomplete) aswell which is also where I got the list of options `flavorOptions.`

**Comment by Mahmoud-zino** [comment]
regarding the animations, I believe it was removed because of performance issues and should be fixed after introducing opt-in transitions #1533 in the future.

The problem with tabbing here is the `focusablePopupElements` in `popup.ts` are not updated when changing the input (updating the autocomplete elements).
an easy fix would be to update the `focusablePopupElements` in `onWindowKeyDown`.

I would be happy to fix it if this was the best solution.

**Comment by endigo9740** [comment]
@Mahmoud-zino is correct on the removal because the component became unresponsive with large numbers of items, due to the transitions. We do plan to reimplement when the dynamic transitions are fully integrated and users can choose to opt in/out at their own discresion.

I'd welcome a fix if you have time!
